instrumentation/aws-lambda/aws-lambda-core-1.0/library/src/main/java/io/opentelemetry/instrumentation/awslambdacore/v1_0/TracingRequestHandler.java

```diff
@@ -98,4 +88,12 @@ public final O handleRequest(I input, Context context) {
   }
 
   protected abstract O doHandleRequest(I input, Context context);
+
+  /**
+   * Returns the HTTP headers for the request extracted from the request handler's input. By
+   * default, returns empty headers.
+   */
+  protected Map<String, String> extractHttpHeaders(I input) {
```

Comment:
This is the most invasive change, since a handler that uses ApiGatewayProxyEvent would need to have its HTTP headers extracted. That being said, headers could be represented by input in other unknown ways so giving a hook into that could be considered generally useful (with limited practical use case).